src/main/java/fruitshop/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package fruitshop.strategy;
+
+import fruitshop.model.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler>
+                                         operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void getOperation(FruitTransaction fruitTransaction) {
+        OperationHandler operationHandler = operationHandlers.get(fruitTransaction.getOperation());
+        operationHandler.handler(fruitTransaction);
```

Comment:
The method name `getOperation` is misleading as it does not return an operation but rather performs an action using the operation. Consider renaming it to reflect the action it performs, such as `executeOperation` or `applyOperation`. This will improve readability and comply with the method naming conventions.